SliceTracker/SliceTracker.py

```diff
@@ -2880,16 +2880,24 @@ def applyRegistration(self, progressCallback=None):
 
     coverProstateRegResult = self.registrationResults.getMostRecentApprovedCoverProstateRegistration()
     lastRigidTfm = self.registrationResults.getLastApprovedRigidTransformation()
+    lastApprovedTfm = self.registrationResults.getMostRecentApprovedTransform()
+    initialTransform = lastApprovedTfm if lastApprovedTfm else lastRigidTfm
 
     self.generateNameAndCreateRegistrationResult(self.currentIntraopVolume)
     parameterNode = slicer.vtkMRMLScriptedModuleNode()
     parameterNode.SetAttribute('FixedImageNodeID', self.currentIntraopVolume.GetID())
-    parameterNode.SetAttribute('FixedLabelNodeID', coverProstateRegResult.fixedLabel.GetID())
+
+    fixedLabel = self.volumesLogic.CreateAndAddLabelVolume(slicer.mrmlScene, self.currentIntraopVolume,
+                                                           self.currentIntraopVolume.GetName() + '-label')
+
+    self.runBRAINSResample(inputVolume=coverProstateRegResult.fixedLabel, referenceVolume=self.currentIntraopVolume,
+                           outputVolume=fixedLabel, warpTransform=initialTransform)
```

Comment:
I don't get it. You are taking the fixed label from coverProstate registration (the label of the first needle image? or previous needle image?) and apply the transformation, which maps coverProstate to the last registered needle image. This doesn't seem to make sense. Did you mean to apply the transform to the movingLabel?